vhdbuilder/packer/pre-install-dependencies.sh

```diff
@@ -102,6 +102,12 @@ else
   # so we just hold the kernel image packages for now on CVM.
   # this still allows us base image and package updates on a weekly cadence.
   if [[ "$IMG_SKU" != "20_04-lts-cvm" ]]; then
+    # Canonical snapshot is only implemented for 20.04 LTS, 22.04 LTS and 23.10 and above
+    # For 20.04, the only SKUs we support are FIPS, and it reaches out to ESM to get the packages, ESM does not have canonical snapshot support
+    # Therefore keeping this to 22.04 only for now
+    if [[ -n "${VHD_BUILD_TIMESTAMP}" && "${OS_VERSION}" == "22.04" ]]; then
+      sed -i "s#http://azure.archive.ubuntu.com/ubuntu/#https://snapshot.ubuntu.com/ubuntu/${VHD_BUILD_TIMESTAMP}#g" /etc/apt/sources.list
```

Comment:
are we duplicating this in install-dependencies.sh?

Reply:
reverting :)

Reply:
comment in install deps

# We dont call out to apt after this in our build process, therefore restoring sources.list back to its original state if we used canonical snapshot earlier
# This will also always happen when cloud-init is applied, therefore, all AKS nodes will always have a fresh sources.list regardless

Reply:
why do we need to do this if we don't ever call out to apt again? and wouldn't we want to always take from the snapshot in all parts of the build?

Reply:
because when the customers get a node from this VHD, we want them to have a regular sources.list with references to ubuntu.archive, snapshot is just for us in our build to have a consistent VHD, once customer gets a node with this VHD, we don't want to impose a snapshot on them.

this pattern is also followed for security patches